channel/src/main/java/com/synopsys/integration/alert/channel/jira/common/util/JiraErrorMessageUtility.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * channel
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.jira.common.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.synopsys.integration.alert.common.descriptor.config.field.errors.AlertFieldStatus;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.exception.AlertFieldException;
+import com.synopsys.integration.rest.exception.IntegrationRestException;
+
+public class JiraErrorMessageUtility {
+    private final Gson gson;
+
+    public JiraErrorMessageUtility(Gson gson) {
+        this.gson = gson;
+    }
+
+    public AlertException improveRestException(IntegrationRestException restException, String issueCreatorFieldKey, String issueCreatorEmail) {
+        String message = restException.getMessage();
+        try {
+            List<String> responseErrors = extractErrorsFromResponseContent(restException.getHttpResponseContent(), issueCreatorFieldKey, issueCreatorEmail);
+            if (!responseErrors.isEmpty()) {
+                message += " | Details: " + StringUtils.join(responseErrors, ", ");
+            }
+        } catch (AlertFieldException reporterException) {
+            return reporterException;
+        }
+        return new AlertException(message, restException);
+    }
+
+    private List<String> extractErrorsFromResponseContent(String httpResponseContent, String issueCreatorFieldKey, String issueCreatorEmail) throws AlertFieldException {
+        JsonObject responseContentObject = gson.fromJson(httpResponseContent, JsonObject.class);
+        if (null != responseContentObject && responseContentObject.has("errors")) {
+            return extractSpecificErrorsFromErrorsObject(responseContentObject.getAsJsonObject("errors"), issueCreatorFieldKey, issueCreatorEmail);
+        }
+        return List.of();
```

Comment:
Do we prefer this of `Collections.emptyList()`?

Reply:
With Java 11 we prefer `List.of()`.